src/node_crypto.cc

```diff
@@ -3349,11 +3348,19 @@ void CipherBase::Init(const char* cipher_type,
                                key_buf_len,
                                1,
                                key,
-                               iv);
+                               nullptr);
 
   EVP_CIPHER_CTX_init(&ctx_);
   const bool encrypt = (kind_ == kCipher);
+
   EVP_CipherInit_ex(&ctx_, cipher_, nullptr, nullptr, nullptr, encrypt);
+
+  const int iv_len = EVP_CIPHER_iv_length(cipher_);
+
+  if (encrypt && iv_len != 0) {
+    return env()->ThrowError("crypto.createCipher() is no longer supported with ciphers that require initialization vectors. Generate a random IV and pass it to crypto.createCipheriv().");
+  }
```

Comment:
Yep, and that's the point: Our deprecation policy *requires* that APIs go through a proper deprecation cycle so this PR would not be able to land as is. The first step is to do a docs or runtime deprecation as a semver-major, which would go into Node.js 9.0.0.